### PR TITLE
[EASI-2407] - Analyze Audits

### PR DESCRIPTION
**Pull request by rileyanderson**
# EASI-2407

## Changes and Description

- Adds `analyzed_audit` table and AnalyzedAudit struct  

  This will analyze a model for a given date (`pkg/worker/analyzed_audit.go`). It will store the the changes in the `changes` 
  JSONb column to use when generating an email from template. This JSON will map to structs found in 
  `pkg/models/analyzed_audit.go`. The JSON is in this format:
  ```json
  {
    "crTdls": {
      "activity": true
    },
    "documents": {
    "count": 1
    },
    "modelPlan": {
      "nameChange": {
        "new": "Test Plan2",
        "old": null
      },
      "statusChanges": [
        "PLAN_DRAFT"
      ]
    },
    "planSections": {
      "updated": [
        "plan_payments",
        "plan_ops_eval_and_learning",
        "plan_beneficiaries",
        "plan_participants_and_providers",
        "plan_general_characteristics",
        "plan_basics"
      ],
      "readyForReview": [
        "plan_payments",
        "plan_ops_eval_and_learning",
        "plan_beneficiaries"
      ],
      "readyForClearance": [
        "plan_participants_and_providers",
        "plan_general_characteristics",
        "plan_basics"
      ]
    },
    "planDiscussion": {
      "activity": true
    },
    "planCollaborators": {
      "added": [
        "New Collaborator",
        "Test User"
      ]
    }
  }
  ```

- Adds a `Worker` struct  in order to share and use dependencies in worker functions.
- Moves the worker setup and start up to `routes.go` in order to use the same store and logger instances.

## How to test this change

There isn't really a good way to test this besides running and modifying tests (TDD FTW). This will be easier to test once we get the rest of Faktory configured and other jobs implemented.

`scripts/dev test:go:only "github.com/cmsgov/mint-app/pkg/worker"`

## PR Author Review Checklist

- [x] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [x] Added or updated tests for backend resolvers or other functions as needed.
- [ ] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [ ] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

### CODE REVIEW

File: pkg/worker/analyzed_audit.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+	"github.com/google/uuid"
+	"github.com/samber/lo"
+
+	"github.com/cmsgov/mint-app/pkg/models"
+)
+
+// AnalyzedAuditJob analyzes the given model and model relations on the specified date
+// args[0] modelPlanID, args[1] date
+func (w *Worker) AnalyzedAuditJob(ctx context.Context, args ...interface{}) error {
```
Comment (rileyanderson) [inline pkg/worker/analyzed_audit.go:15]: `context.Context` and sending other arguments in `args ...interface{}` is required for Faktory Jobs.

File: pkg/worker/worker_test_utilities.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+		Port:           config.GetString(appconfig.DBPortConfigKey),
+		Database:       config.GetString(appconfig.DBNameConfigKey),
+		Username:       config.GetString(appconfig.DBUsernameConfigKey),
+		Password:       config.GetString(appconfig.DBPasswordConfigKey),
+		SSLMode:        config.GetString(appconfig.DBSSLModeConfigKey),
+		MaxConnections: config.GetInt(appconfig.DBMaxConnections),
+	}
+}
+
+func getTestDependencies() (storage.DBConfig, *ld.LDClient, *zap.Logger, *models.UserInfo, *pubsub.ServicePubSub) {
```
Comment (StevenWadeOddball) [inline pkg/worker/worker_test_utilities.go:86]: Do these need to be here in this file? Could you import resolver for these? I see they are basically the same as in resolver_test utilities.

**Review by StevenWadeOddball — APPROVED** [review]
Overall looks great! A couple non-blocking comments. It would be nice if we didn't need to have another test utility file, but I can imagine that import cycle dependencies could necessitate that.
Thanks Riley!

File: .vscode/settings.json [hunk 3]
```diff
@@ -3,4 +3,4 @@
     "editor.codeActionsOnSave": {
         "source.fixAll.eslint": true
     }
-}
```
[the remark this answers is not in the document]
Comment (rileyanderson) [reply .vscode/settings.json:6]: Probably my editor. Ideally every file should end in a newline (most do in the project)

File: pkg/worker/worker_test_utilities.go (hunk 2 above)
Comment (rileyanderson) [reply pkg/worker/worker_test_utilities.go:86]: Ah yeah probably. Let me check

Comment (rileyanderson) [reply pkg/worker/worker_test_utilities.go:86]: @StevenWadeOddball My original thinking was maybe once we start testing more Faktory related stuff we might need some additional config/setup and that the Worker is a separate entity from the Server but I see what you're saying, it is pretty similar right now.

**Pull request merged**